createdisk.sh

```diff
@@ -78,9 +78,9 @@ EOF
     # also in case of microshift the ports like 2222, 443, 80 ..etc need to be manually added
     # and OCP/OKD/podman bundles have it disabled by default.
     ${SSH} core@${VM_IP} -- sudo systemctl disable firewalld
-    # Copy the sample microshift config and update the base domain with crc base domain
-    ${SSH} core@${VM_IP} -- sudo cp /etc/microshift/config.yaml.default /etc/microshift/config.yaml
-    ${SSH} core@${VM_IP} -- "sudo sed -i 's/#baseDomain: .*/baseDomain: ${SNC_PRODUCT_NAME}.${BASE_DOMAIN}/g' /etc/microshift/config.yaml"
+    ${SSH} core@${VM_IP} -- "sudo podman run --rm  -v /etc/microshift:/workdir:Z docker.io/mikefarah/yq:4.35.2 eval '.dns.baseDomain = \"${SNC_PRODUCT_NAME}.${BASE_DOMAIN}\"'  config.yaml.default | sudo tee /etc/microshift/config.yaml"
```

Comment:
Do we need to run `podman` as root? I don't think `--rm` will remove the container image which was just pulled.
I'd prefer that we don't add a new external dependency to our build process, mostly for security reasons.

Reply:
ah I need to remove the image, will update but if we remove it then there is no external dep left around also on the host we download the `yq` binary from github.com/mikefarah/yq and same we are doing as container image.

Reply:
> Do we need to run podman as root?

Does it even matter in this command context? root or rootless at the end we want to have update the config file and delete it and then remove the image.

Reply:
Also with `rooless` it is not work because we are mounting `/etc/microshift` which have `root` ownership so it fail with `lsetxattr /etc/microshift/config.yaml.default: operation not permitted`

Reply:
> also on the host we download the yq binary from github.com/mikefarah/yq and same we are doing as container image.

The `yq` binary and this container image are 2 different "binaries", yes we already use a linux binary from a given github.com/mikefarah/yq release, but docker.io/mikefarah/yq:4.35.2 is another different binary that we decide to use and trust. 

> > Do we need to run podman as root?
> 
> Does it even matter in this command context? root or rootless at the end we want to have update the config file and delete it and then remove the image.

We run an untrusted container image inside the VM we are going to ship to customers. Imo yes it matters, we should think twice before running untrusted binaries as root inside this VM. And the less we do this, the better.

Reply:
> The `yq` binary and this container image are 2 different "binaries", yes we already use a linux binary from a given github.com/mikefarah/yq release, but docker.io/mikefarah/yq:4.35.2 is another different binary that we decide to use and trust.

If that is the concern then I can use same latest release for both the places.

> We run an untrusted container image inside the VM we are going to ship to customers. Imo yes it matters, we should think twice before running untrusted binaries as root inside this VM. And the less we do this, the better.

https://github.com/mikefarah/yq/blob/master/Dockerfile is what this image built, I am not sure when you say untrusted what you mean, some random image which we don't even sure how it built or even the images which we have access to container file and the source also?

Reply:
We have no control over docker.io/mikefarah, I don't think we have a way of verifying that the container that you pull was really built using the docker file you linked to, `FROM alpine:3` in the file that you linked is also questionable, no way of knowing what exactly the base image was.

In short, this is one new piece of untrusted code that we are going to use in our build process.

Reply:
I don't wanted to block this PR over this dockerfile argument and since it really need for release I went with using YQ binary over host.